fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadMgr.java

```diff
@@ -448,57 +448,46 @@ public RoutineLoadJob getJob(String dbFullName, String jobName) throws MetaNotFo
         }
     }
 
-    /*
-      if dbFullName is null, result = all of routine load job in all of db
-      else if jobName is null, result =  all of routine load job in dbFullName
-
-      if includeHistory is false, filter not running job in result
-      else return all of result
+    /**
+     * use dbFullName and jobName to filter routine load jobs.
+     * if includeHistory is false, filter not running job in result else return all of result.
      */
     public List<RoutineLoadJob> getJob(String dbFullName, String jobName, boolean includeHistory)
             throws MetaNotFoundException {
+        List<RoutineLoadJob> result = Lists.newArrayList();
         readLock();
         try {
-            // return all of routine load job
-            List<RoutineLoadJob> result;
-            RESULT:
-            {
-                if (dbFullName == null) {
-                    result = new ArrayList<>(idToRoutineLoadJob.values());
-                    sortRoutineLoadJob(result);
-                    break RESULT;
-                }
-
+            if (dbFullName == null && jobName == null) {
+                result.addAll(idToRoutineLoadJob.values());
+                sortRoutineLoadJob(result);
```

Comment:
is it possible to sort out of the lock?

Reply:
sort is by `job state`, better to do it in the lock.